File: frontend/src/types/payment.ts
export interface Payment {
    id: string;
    amount: number;
    paidAt: string; // ISO string from backend
    debtId: string;
}

export interface PaymentApi {
    id: string;
    amount: number;
    paidAt: string;
    debt: {
        id: string;
        name: string;
    };
    originalAmount: number;
    minPayment: number;
    interestRate: number;
}

File: backend/Models/Payment.cs
namespace backend.Models;

public class Payment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public decimal Amount { get; set; }
    public DateTime PaidAt { get; set; } = DateTime.UtcNow;

    // relationship

    public Guid DebtId { get; set; }
    public Debt Debt { get; set; } = null!;
}
